gattc_darwin.go

```diff
@@ -145,3 +152,20 @@ func (c DeviceCharacteristic) EnableNotifications(callback func(buf []byte)) err
 
 	return nil
 }
+
+// Read reads the current characteristic value.
+func (c *deviceCharacteristic) Read() (data []byte, err error) {
+	c.readChan = make(chan error)
+	c.service.device.prph.ReadCharacteristic(c.characteristic)
+
+	// wait for result
+	select {
+	case <-c.readChan:
+		c.readChan = nil
+	case <-time.NewTimer(10 * time.Second).C:
+		c.readChan = nil
+		return nil, errors.New("timeout on Read()")
```

Comment:
While it might not be a bad idea, this should be unnecessary: CoreBluetooth should return an error instead of waiting forever.

Reply:
Probably true, but still seems like a good idea.